src/Collections/AbstractCollection.php

```diff
@@ -9,12 +9,12 @@ class AbstractCollection implements MultiFieldsEntityInterface
 {
     private int $pointer = 0;
     private array $items;
-
+    protected string $property_name;
     /**
      * @template T
-     * @var T $classname
+     * @var class-string T $classname
      */
-    public function __construct(mixed $classObj)
+    public function __construct(mixed $class_string, $property_name)
```

Comment:
It appears that in some places you are using camelCase and in other places it's snake_case. I would use the community PSR coding standards.
https://www.php-fig.org/psr/psr-12/

Reply:
Think I can update text using PSR standards :)